cmd/scale.go

```diff
@@ -217,14 +217,36 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 
 	ctx, cancel := context.WithTimeout(context.Background(), armhelpers.DefaultARMOperationTimeout)
 	defer cancel()
+	if sc.containerService.Properties.MasterProfile != nil && sc.containerService.Properties.MasterProfile.IsAvailabilitySets() ||
+		sc.agentPool.IsAvailabilitySets() {
+		availabilitySetIDs := []string{}
+		for vmsListPage, err := sc.client.ListVirtualMachines(ctx, sc.resourceGroupName); vmsListPage.NotDone(); err = vmsListPage.Next() {
+			if err != nil {
+				return errors.Wrap(err, "failed to get vms in the resource group")
```

Comment:
I think it would be more defensive and safer just to log these VM list errors instead of failing.